### DISCUSSION TITLE
Bootstrap Blazor 使用模板创建项目

### DISCUSSION THREAD

**Discussion by YSMC-W** [dotnetcore/BootstrapBlazor#307] [Quick start] [answered]
原文连接：https://www.cnblogs.com/ysmc/p/16101157.html

BootstrapBlazor 官网地址：https://www.blazor.zone/

　　BootstrapBlazor 提供了一个项目模板，以便大家快速创建项目，废话不多说；

模板安装
　　模板安装分为两种方式，手动下载插件安装 与 命令行安装

1、手动下载安装插件
　　首先到微软插件中心下载项目模板，[传送门](https://marketplace.visualstudio.com/items?itemName=Longbow.BootstrapBlazorUITemplate)，同时下方也是有对应的教程的，获取可以到项目的开源库上查看Wiki，本项目托管在Gitee上，Wiki [传送门](https://gitee.com/LongbowEnterprise/BootstrapBlazor/wikis/%E9%A1%B9%E7%9B%AE%E6%A8%A1%E6%9D%BF%E4%BD%BF%E7%94%A8%E6%95%99%E7%A8%8B)

目前最新的项目模板 已升级到 net6 仅支持 visual studio 2022，这点请大家注意；

![image](https://user-images.githubusercontent.com/15371924/209424050-d338a350-e117-4c93-b082-12c79f439622.png)

　　下载完成后双击打开文件，进行安装，进行前请先退出掉 vVsual Studio 

![image](https://user-images.githubusercontent.com/15371924/209424163-54830007-7e54-4e05-9ed2-0b6ed51eeb23.png)

 点击 Install，等待安装完成即可，这里可能会出现卡很久，或者直接卡死的情况，这个是微软的bug，呼出 任务管理器 关掉重新来就可以了；

![image](https://user-images.githubusercontent.com/15371924/209424080-417cba7b-b9e0-42bc-a27d-b796fabfd0a9.png)

 2、命令行安装
　　相对于手动下载模板插件方式，命令行明显简单太多， 一行命令就完事了

`
dotnet new -i Bootstrap.Blazor.Templates::7.0.0 #这里可指定模板的版本号
`

 使用模板创建项目
　　打开 Visual Studio 2022，点击创建新项目，选择刚刚安装好的 Bootstrap Blazor Project Template 模板

![image](https://user-images.githubusercontent.com/15371924/209424132-d0f62c9b-ee0a-4dfb-9add-9977a2adc8c4.png)
![image](https://user-images.githubusercontent.com/15371924/209424139-e2b10e29-d08a-4125-af60-ce08dfb66aae.png)

 填写好项目信息后，点击创建就可以啦
![image](https://user-images.githubusercontent.com/15371924/209424143-f7e32b05-96a9-4162-a91e-c4ad52c21e84.png)

 这是模板创建好的解决方案目录结构，建议组件都写在 Shared 工程里，方便兼容 SSR 与 Wasm 两种模式，模板都已经为你创建好了！

![image](https://user-images.githubusercontent.com/15371924/209424146-9ffe2382-ce2c-4d81-94f8-79a0ba66ecda.png)

 运行项目
　　直接无脑 F5 ，我们先一睹为快
![image](https://user-images.githubusercontent.com/15371924/209424149-a1b327bc-ac68-4088-8217-9dbf59df2957.png)

  可以看到，基本的一个项目就出来， 有左侧的菜单，右上角的消息组件，是不是非常的方便！而且不要忽略了右上角的设置按钮，它提供了各种的设置，满足你的项目所需
![image](https://user-images.githubusercontent.com/15371924/209424152-f3598bed-5d51-4d88-b818-21ff44e0f227.png)

 好了，至此，你就可以愉快的开发了，是不是很爽！

**Comment by densen2014** [comment, marked as answer]
来啦
